core/types/decimal/decimal.go

```diff
@@ -127,7 +127,7 @@ func inferPrecisionAndScale(s string) (precision, scale uint16) {
 
 	intPart := uint16(len(parts[0]))
 	if len(parts) == 1 {
-		return intPart, 0
+		return max(1, intPart), 0 // at least 1 for precision with 0 scale
```

Comment:
In `utils/conv/conv_test.go` I discovered that `"0"` could not be used as a valid Decimal.  I don't think that is intended, and this case catches if `strings.TrimLeft(parts[0], "0")` reduced it to an empty string.  @brennanjl lmk if a zero decimal is actually not supposed to be allowed.  Based on code in other places, it looks like it should have been.